src/helpers.rs

```diff
@@ -846,6 +864,18 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         self.set_last_error(self.io_error_to_errnum(err)?)
     }
 
+    /// Helper function that consumes an `std::io::Result<i32>` and writes the `Ok` integer
+    /// to the destination. In case the result is an error, this function writes
+    /// `-1` and sets the last OS error accordingly.
+    fn write_io_result(
```

Comment:
Please move this to the new io_error.rs file, and put it right below `try_unwrap_io_result` (since it extends that helper).